.github/workflows/pr_checks.yaml

```diff
@@ -0,0 +1,120 @@
+name: PR Checks
+
+on:
+  push:
+    branches:
+      - "main"
+  pull_request:
+    branches:
+      - "**"
+
+jobs:
+  tests:
+    name: Tests
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Set up Python 3.10
+        uses: actions/setup-python@v5
+        with:
+          python-version: "3.10"
+
+      - name: Set up Environment
+        id: setup
+        run: |
+          curl -LsSf https://astral.sh/uv/install.sh | sh
+          uv venv
+          echo "VIRTUAL_ENV=$HOME/.venv" >> $GITHUB_ENV
+        continue-on-error: false
+
+      - name: Install Package
+        id: install_package
+        run: |
+          . .venv/bin/activate
+          uv pip install "lm_buddy[dev] @ ."
+        continue-on-error: false
+
+      - name: Unit Tests
+        id: unit_tests
+        run: |
+          . .venv/bin/activate
+          pytest tests/unit
+        continue-on-error: false
+
+      - name: Integration Tests
```

Comment:
would this mean that we run integration tests twice times when we merge into main as this is activated since we would now have two workflows that run against main?

Reply:
oh, just saw we deleted the main workflow here and are replacing, ignore